### PR TITLE
make Netweaver PAS/AAS virtual IPs reachable on public cloud setups

### PR DESCRIPTION
**Pull request by yeoldegrove**
The Netweaver virtual IPs for AS/ERS are reachable via different mechanism for public clouds:
- azure -> load-balancer
- gcp -> route to VM (changed by cluster)
- aws -> route to VM (changed by cluster)

For the Netweaver virtual IPs for PAS/AAS there is currently no mechanism implemented to make them reachable to other nodes.
Therefore the current monitoring setup #740 is also broken.

This PR makes these PAS/AAS virtual IPs reachable via these mechanisms:
- azure -> vIP on network interface on VMs
  - load-balancer setup would also be possible in theory but is way more complex (e.g. create rules per node)
- gcp -> route to VM (static)
- aws -> route to VM (static)

In addition to that there are fixes included to not only deploy AS routes in HA scenarios.
```
count = local.create_ha_infra
vs.
count = local.vm_count > 0 ? 1 : 0
```

### CODE REVIEW

File: aws/modules/netweaver_node/main.tf [hunk 1]
```diff
@@ -22,20 +23,36 @@ resource "aws_route_table_association" "netweaver-subnet-route-association" {
...
-resource "aws_route" "ers-cluster-vip" {
+resource "aws_route" "nw-ers-route" {
   count                  = local.create_ha_infra
   route_table_id         = var.route_table_id
   destination_cidr_block = "${element(var.virtual_host_ips, 1)}/32"
   instance_id            = aws_instance.netweaver.1.id
 }
 
+# deploy if only PAS
+resource "aws_route" "nw-pas-route" {
```
Comment (arbulu89) [inline aws/modules/netweaver_node/main.tf:41]: I would put a comment to say that this scenario is when the PAS is installed in the same machine than the ASCS

File: azure/modules/netweaver_node/main.tf [hunk 2]
```diff
@@ -210,12 +211,48 @@ resource "azurerm_network_interface" "netweaver" {
...
   ip_configuration {
     name                          = "ipconf-primary"
+    primary                       = true
     subnet_id                     = var.network_subnet_id
     private_ip_address_allocation = "static"
     private_ip_address            = element(var.host_ips, count.index)
     public_ip_address_id          = local.bastion_enabled ? null : element(azurerm_public_ip.netweaver.*.id, count.index)
   }
 
+  # deploy if non-HA AS
```
Comment (arbulu89) [inline azure/modules/netweaver_node/main.tf:221]: Is this the ASCS?

File: gcp/modules/netweaver_node/main.tf [hunk 3]
```diff
@@ -37,6 +38,28 @@ resource "google_compute_route" "nw-ers-route" {
...
+  dest_range             = "${element(var.virtual_host_ips, local.app_start_index)}/32"
+  network                = var.network_name
+  next_hop_instance      = google_compute_instance.netweaver.0.name
+  next_hop_instance_zone = element(var.compute_zones, 0)
+  priority               = 1000
+}
+
+# deploy if PAS and AAS
+resource "google_compute_route" "nw-app-route" {
+  name                   = "${var.common_variables["deployment_name"]}-nw-appoute-${format("%02d", local.app_start_index + count.index + 1)}"
```
Comment (arbulu89) [inline gcp/modules/netweaver_node/main.tf:54]: Typo here? `appoute`

**Review by arbulu89 — APPROVED** [review]
@eik
The changes look correct so far, but I have added some small comments.

File: aws/modules/netweaver_node/main.tf (hunk 1 above)
Comment (yeoldegrove) [reply aws/modules/netweaver_node/main.tf:41]: I made the comments more clear.

File: gcp/modules/netweaver_node/main.tf (hunk 3 above)
Comment (yeoldegrove) [reply gcp/modules/netweaver_node/main.tf]: Yes, named it that way now.

File: azure/modules/netweaver_node/main.tf (hunk 2 above)
Comment (yeoldegrove) [reply azure/modules/netweaver_node/main.tf]: Yes, named it that way now.

**Pull request merged**